src/services/dashboard-service.js

```diff
@@ -136,6 +136,34 @@ class DashboardService {
 
     return activities;
   }
+
+  async getUserFavorites(userId) {
+    const user = await this.userStore.getUserById(userId);
+
+    const documentIds = user.favorites.filter(f => f.type === FAVORITE_TYPE.document).map(d => d.id);
+    const roomIds = user.favorites.filter(f => f.type === FAVORITE_TYPE.room).map(r => r.id);
+    const lessonIds = user.favorites.filter(f => f.type === FAVORITE_TYPE.lesson).map(l => l.id);
+
+    const documents = documentIds.length ? await this.documentStore.getDocumentsMetadataByKeys(documentIds) : [];
+    const rooms = roomIds.length ? await this.roomStore.getRoomsByIds(roomIds) : [];
+    const lessons = lessonIds.length ? await this.lessonStore.getLessonsMetadataByIds(lessonIds) : [];
```

Comment:
We could parallelize this by `Promise.all`

Reply:
Indeed, done, thanks!

Reply:
:+1: